R/glmClassical.R

```diff
@@ -0,0 +1,1199 @@
+#
+# Copyright (C) 2013-2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#still to do: multinomial, ordinal, negative binomial, quasi
+
+GeneralizedLinearModel <- function(jaspResults, dataset = NULL, options, ...) {
+  if (options$family == "binomial") {
+    ready <- (options$dependent != "" && options$weights != "")
+
+  } else {
+    ready <- options$dependent != ""
+  }
+
+
+  if (ready) {
+    dataset <- .glmReadData(dataset, options)
+    .glmCheckDataErrors(dataset, options)
+  }
+
+  #output tables
+  .glmModelSummaryTable(jaspResults, dataset, options, ready, position = 1)
+  .glmModelFitTable(    jaspResults, dataset, options, ready, position = 2)
+  .glmEstimatesTable(   jaspResults, dataset, options, ready, position = 3)
+
+  #diagnostic tables and plots
+  .glmPlotResVsFitted(jaspResults, dataset, options, ready, position = 4)
+
+  .glmPlotResVsPredictor(jaspResults, dataset, options, ready, residType = "deviance", position = 5)
+  .glmPlotResVsPredictor(jaspResults, dataset, options, ready, residType = "Pearson", position = 6)
+  .glmPlotResVsPredictor(jaspResults, dataset, options, ready, residType = "quantile", position = 7)
+
+  .glmPlotResQQ(jaspResults, dataset, options, ready, position = 8)
+
+  .glmPlotResPartial(jaspResults, dataset, options, ready, position = 9)
+
+  .glmPlotZVsEta(jaspResults, dataset, options, ready, position = 10)
+
+  .glmOutlierTable(jaspResults, dataset, options, ready, position = 11, residType = "quantile")
+  .glmOutlierTable(jaspResults, dataset, options, ready, position = 11, residType = "standardized deviance")
+  .glmOutlierTable(jaspResults, dataset, options, ready, position = 11, residType = "studentized deviance")
+
+  .glmInfluenceTable(jaspResults, dataset, options, ready, position = 12)
+  .glmMulticolliTable(jaspResults, dataset, options, ready, position = 13)
```

Comment:
Could you move this into a separate JASP container? This means you can make a `.glmDiagnostics` function that would simplify the main function + the output is more structured (i.e., all diagnostic plots and tables are under one "Diagnostics" heading).

Reply:
Will do.